### PR TITLE
Upgrade to rdkafka ruby

### PR DESCRIPTION
**Pull request by binarycleric**
Working to upgrade the demo app from using kafka-ruby to rdkafka-ruby.
Configuration is still not correct and the message consumer isn't working yet.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -9,51 +9,41 @@
...
-    ssl_client_cert_key: ENV.fetch('KAFKA_CLIENT_CERT_KEY'),
-    ssl_verify_hostname: false,
-  )
+  $producer = Rdkafka::Config.new({
+    :"bootstrap.servers" => ENV.fetch('KAFKA_URL').gsub('kafka+ssl://', ''),
+    :"security.protocol" => "ssl",
+    :"ssl.ca.location" => tmp_ca_file.path,
+    :"ssl.key.pem" => ENV.fetch('KAFKA_CLIENT_CERT_KEY'),
+    :"ssl.certificate.pem" => ENV.fetch('KAFKA_CLIENT_CERT'),
+    :"enable.ssl.certificate.verification" => false,
```
Comment (ekozilforce) [inline app.rb:23]: Do we want to verify ssl certs in this poc?

File: app.rb [hunk 2]
```diff
@@ -9,51 +9,41 @@
 KAFKA_TOPIC = ENV.fetch('KAFKA_TOPIC', 'messages')
 GROUP_ID = ENV.fetch('KAFKA_CONSUMER_GROUP', 'heroku-kafka-demo')
 
-def with_prefix(name)
-  "#{ENV['KAFKA_PREFIX']}#{name}"
-end
```
Comment (mble-sfdc) [inline app.rb:14]: Why was this removed? `KAFKA_PREFIX` is still required for the topic and consumer group.

File: app.rb [hunk 3]
```diff
@@ -9,51 +9,41 @@
...
+    :"group.id" => GROUP_ID,
+  }).consumer
+
   $recent_messages = []
   start_consumer
-  start_metrics
 
   at_exit do
-    $producer.shutdown
+    $producer.close
```
Comment (mble-sfdc) [inline app.rb:46]: `$producer.flush` should be called before `$producer.close` to ensure all messages are sent to Kafka before closing the producer.

Comment (binarycleric) [reply app.rb:51]: Thanks. Updated.

File: app.rb (hunk 2 above)
Comment (binarycleric) [reply app.rb]: That was a mistake on my part and it has been re-added.

**Review by troycoll — APPROVED** [review]
👍 👍

**Pull request merged**